src/BenchmarkDotNet/Running/BenchmarkRunnerDirty.cs

```diff
@@ -143,5 +144,66 @@ private static Summary[] RunWithExceptionHandling(Func<Summary[]> run)
                 return new[] { Summary.NothingToRun(e.Message, string.Empty, string.Empty) };
             }
         }
+
+        private static BenchmarkRunInfo[] TypeToBenchmarks(Type type, IConfig config)
+        {
+            if (type is null)
+                throw new InvalidBenchmarkDeclarationException("Type not provided.");
+
+            if (!type.ContainsRunnableBenchmarks())
+                throw new InvalidBenchmarkDeclarationException($"{type.Name} must be public non-sealed non-static type and contain public [Benchmark] methods.");
+
+            return new[] { BenchmarkConverter.TypeToBenchmarks(type, config) };
+        }
+
+        private static BenchmarkRunInfo[] MethodsToBenchmarks(Type containingType, MethodInfo[] methods, IConfig config)
+        {
+            if (containingType is null)
+                throw new InvalidBenchmarkDeclarationException("Type not provided.");
+
+            if (!containingType.ContainsRunnableBenchmarks())
+                throw new InvalidBenchmarkDeclarationException($"{containingType.Name} must be public non-sealed non-static type and contain public [Benchmark] methods.");
+
+            if (methods.IsNullOrEmpty())
+                throw new InvalidBenchmarkDeclarationException($"No methods provided for {containingType.Name} type.");
+
+            if (methods.Any(m => m is null))
+                throw new InvalidBenchmarkDeclarationException($"Null not allowed for benchmark methods.");
+
+            var publicBenchmarkMethodsOfType = containingType.GetMethods().Where(m => m.HasAttribute<BenchmarkAttribute>()).ToArray();
+            var wrongMethods = methods.Except(publicBenchmarkMethodsOfType).ToArray();
+            if (!wrongMethods.IsEmpty())
+                throw new InvalidBenchmarkDeclarationException(string.Join(", ", wrongMethods.Select(m => m.Name)) + 
+                                                               $" are wrong. Methods should be with [Benchmark] attribute of {containingType} type.");
```

Comment:
nit: I would add "annotated" here

```suggestion
                                                               $" are wrong. Methods should be annotated with [Benchmark] attribute of {containingType} type.");
```

Reply:
There are already several different versions of this message. I found the most powerful (longest) of them in the library and used it.